components/brave_ads/core/public/ads_callback.h

```diff
@@ -29,14 +30,14 @@ using GetStatementOfAccountsCallback =
     base::OnceCallback<void(mojom::StatementInfoPtr mojom_statement)>;
 
 using MaybeServeNewTabPageAdCallback =
-    base::OnceCallback<void(const std::optional<NewTabPageAdInfo>& ad)>;
+    base::OnceCallback<void(base::optional_ref<const NewTabPageAdInfo> ad)>;
```

Comment:
I'm trying to find some guidance on this... I worry this could be a source of dangling references. It is not clear to me yet if the closure preserves the value in its storage when you have one for something like this

```cxx
base::OnceCallback<void(base::optional_ref<const Foo> foo)>;
```

As opposed to this:
```cxx
base::OnceCallback<void(std::optional<Foo>& foo)>;
```

As I understand in this second case the closure takes a copy of the whole thing, but I'm not sure, so I'm trying to confirm.